### PR TITLE
IPFS-Swarm-choice

### PR DESCRIPTION
**Pull request by dlaxcess**

### CODE REVIEW

File: common/lib/common/config.ts [hunk 1]
```diff
@@ -412,6 +412,29 @@ const explorerTxLink = (chainId: number, tx: string): string =>
...
+const uriShort = (uri: string) => {
+  const [storage, hash] = uri.split("://");
+  if (!(storage && hash && hash.length > 20)) return uri;
+
+  return `${storage}://${hash.slice(0, 8)}...${hash.slice(-8)}`;
+};
+
+const uriToUrl = (uri: string) => {
+  const [storage, hash] = uri.split("://");
+  if (!(storage && hash && hash.length > 20)) return uri;
```
Comment (zapaz) [inline common/lib/common/config.ts:427]: remove` test > 20`, not mandatory here

File: wordpress/kredeum-nfts.php_handelbars [hunk 2]
```diff
@@ -69,12 +72,18 @@ if ( is_admin() ) {
 	);
 }
 
-define( 'IPFS_GATEWAY', 'https://ipfs.io/ipfs/' );
-define( 'NFT_STORAGE_KEY', get_option( 'NFT_STORAGE_KEY', '' ) ? get_option( 'NFT_STORAGE_KEY', '' ) : 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiJkaWQ6ZXRocjoweEVkYzMzMjZEODc3OGU5QWVhMGZGMzc0MmEyRERGRDREM0E0NTYzNTciLCJpc3MiOiJuZnQtc3RvcmFnZSIsImlhdCI6MTYyMTAwMjI3ODIzNywibmFtZSI6Im9wZW4tbmZ0cyJ9.EJeI37e2D9k09coXLIz8IgqFi85gHlLteE-ETMcRt8I' );
+// IPFS.
+define( 'IPFS_GATEWAY', '{{storage.ipfs.gateway}}' );
+define( 'NFT_STORAGE_KEY', get_option( '_KRE_NFT_STORAGE_KEY', '' ) ? get_option( '_KRE_NFT_STORAGE_KEY', '' ) : 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiJkaWQ6ZXRocjoweEVkYzMzMjZEODc3OGU5QWVhMGZGMzc0MmEyRERGRDREM0E0NTYzNTciLCJpc3MiOiJuZnQtc3RvcmFnZSIsImlhdCI6MTYyMTAwMjI3ODIzNywibmFtZSI6Im9wZW4tbmZ0cyJ9.EJeI37e2D9k09coXLIz8IgqFi85gHlLteE-ETMcRt8I' );
```
Comment (zapaz) [inline wordpress/kredeum-nfts.php_handelbars:77]: replace key with `{{storage.ipfs.nftstoragekey}}`

File: wordpress/plugins/kredeum-nfts/admin/media/post.php [hunk 3]
```diff
@@ -1,43 +1,43 @@
...
+ * affiche box avec URI
  */
 add_action(
 	'add_meta_boxes_attachment',
 	function () {
 		add_meta_box(
-			'ipfs_link_box',
-			'IPFS',
+			'storage_link_box',
+			'DEC STORAGE',
```
Comment (zapaz) [inline wordpress/plugins/kredeum-nfts/admin/media/post.php:19]: `DEC STORAGE` ? => `STORAGE`

File: wordpress/plugins/kredeum-nfts/uninstall.php [hunk 4]
```diff
@@ -10,15 +10,17 @@
...
 
 
 /**
  * Clear settings
-*/
-delete_option( 'ipfs_auto' );
-delete_option( 'kredeum_beta' );
-delete_option( 'nft_storage_key' );
-delete_option( 'select_collection' );
+ */
```
Comment (zapaz) [inline wordpress/plugins/kredeum-nfts/uninstall.php:20]: replace with

```php
$wpdb->query("DELETE FROM $wpdb->options WHERE option_name LIKE '_kre_%'");
```

File: wordpress/kredeum-nfts.php_handelbars (hunk 2 above)
Comment (zapaz) [reply wordpress/kredeum-nfts.php_handelbars:77]: and put the key in config.handlebars.json

**Review by zapaz — CHANGES_REQUESTED** [review]
some small changes

File: common/lib/common/config.ts (hunk 1 above)
Comment (dlaxcess) [reply common/lib/common/config.ts:427]: done

File: wordpress/kredeum-nfts.php_handelbars (hunk 2 above)
Comment (dlaxcess) [reply wordpress/kredeum-nfts.php_handelbars:77]: done

File: wordpress/plugins/kredeum-nfts/admin/media/post.php (hunk 3 above)
Comment (dlaxcess) [reply wordpress/plugins/kredeum-nfts/admin/media/post.php:19]: done

File: wordpress/plugins/kredeum-nfts/uninstall.php (hunk 4 above)
Comment (dlaxcess) [reply wordpress/plugins/kredeum-nfts/uninstall.php:20]: done

**Pull request merged**